File: java/026 Remove Duplicates from Sorted Array/RemoveDuplicatesfromSortedArray.java

/**
 * Given a sorted array, remove the duplicates in place such that each element appear only once and return the new length.
 * Do not allocate extra space for another array, you must do this in place with constant memory.
 */
public class RemoveDuplicatesfromSortedArray {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0)
            return 0;
        int val = nums[0], size = 1;
        for (int i : nums)
            if (i != val)
                nums[size++] = val = i;
        return size;
    }
}

File: kt/026 Remove Duplicates from Sorted Array/RemoveDuplicatesfromSortedArray.kt

/**
 * Given a sorted array, remove the duplicates in place such that each element appear only once and return the new length.
 * Do not allocate extra space for another array, you must do this in place with constant memory.
 */
 class RemoveDuplicatesfromSortedArray {
 	fun removeDuplicates(nums: IntArray): Int {
 		if (nums.size == 0)
 			return 0
 		var value = nums[0]
 		var size = 1
 		for (i in nums)
 			if (i != value) {
 				value = i
 				nums[size++] = value
 			}
 		return size
 	}
 }
k
